lighthouse-cli/test/smokehouse/smokehouse.js

```diff
@@ -0,0 +1,199 @@
+    }
+
+    const actualScore = actualResult.score;
+    const expectedScore = expected.audits[auditName];
+    return {
+      category: auditName,
+      actual: actualScore,
+      expected: expectedScore,
+      equal: actualScore === expectedScore
+    };
+  });
+
+  return {
+    finalUrl: {
+      category: 'final url',
+      actual: actual.url,
+      expected: expected.url,
+      equal: actual.url === expected.url
+    },
+    audits: collatedAudits
+  };
+}
+
+/**
+ * Log the result of an assertion of actual and expected results.
+ * @param {{category: string, equal: boolean, actual: boolean, expected: boolean}} assertion
+ */
+function reportAssertion(assertion) {
+  if (assertion.equal) {
+    console.log(`  ${GREEN_CHECK} ${assertion.category}: ` +
+        greenify(assertion.actual));
+  } else {
+    console.log(`  ${RED_X} ${assertion.category}: ` +
+        redify(`found ${assertion.actual}, expected ${assertion.expected}`));
+  }
+}
+
+/**
+ * Log all the comparisons between actual and expected test results, then print
+ * summary. Returns count of passed and failed tests.
+ * @param {{finalUrl: !Object, audits: !Array<!Object>}} results
+ * @return {{passed: number, failed: number}}
+ */
+function report(results) {
+  reportAssertion(results.finalUrl);
+
+  let correctCount = 0;
+  let failedCount = 0;
+  results.audits.forEach(auditAssertion => {
+    if (auditAssertion.equal) {
+      correctCount++;
+    } else {
+      failedCount++;
+      reportAssertion(auditAssertion);
+    }
+  });
+
+  const plural = correctCount === 1 ? '' : 's';
+  const correctStr = `${correctCount} audit${plural}`;
+  const colorFn = correctCount === 0 ? redify : greenify;
+  console.log(`  Correctly passed ${colorFn(correctStr)}\n`);
+
+  return {
+    passed: correctCount,
+    failed: failedCount
+  };
+}
+
+const cli = yargs
+  .help('help')
+  .describe({
+    'config-path': 'The path to the config JSON file',
+    'expectations-path': 'The path to the expected audit results file'
+  })
+  .default('config-path', DEFAULT_CONFIG_PATH)
+  .default('expectations-path', DEFAULT_EXPECTATIONS_PATH)
+  .argv;
+
+const configPath = resolveLocalOrCwd(cli['config-path']);
+const expectations = require(resolveLocalOrCwd(cli['expectations-path']));
+
+// Loop sequentially over expectations, comparing against Lighthouse run, and
+// reporting result.
+let passingCount = 0;
+let failingCount = 0;
+expectations.forEach(expected => {
+  console.log(`Checking '${expected.initialUrl}'...`);
+  const results = runLighthouse(expected.initialUrl, configPath);
+  const collated = collateResults(results, expected);
+  const counts = report(collated);
+  passingCount += counts.passed;
+  failingCount += counts.failed;
+});
+
+if (passingCount) {
+  console.log(greenify(`${passingCount} passing`));
+}
+if (failingCount) {
+  console.log(redify(`${failingCount} failing`));
+  process.exit(1);
+}
```

Comment:
> a nice next step might be adding support for regression testing things like TTI and FMP

yeah, I intentionally dodged that until we get better known bounds on our test variance so we know what a test regression looks like